### PR TITLE
fix(react): component wrappers work with onDoubleClick

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally for affected output targets
- [x] Tests (`npm test`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Generated React component wrappers will not fire the `dblclick` event when `onDoubleClick` is bound to the wrapper.

e.g.:
```tsx
<MyComponent onDoubleClick={() => alert('Hello world')}>Click me</MyComponent>
```

Issue URL: Relates to: https://github.com/ionic-team/ionic-framework/issues/21320

## What is the new behavior?

- Transforms `doubleclick` naming syntax from React to the browser event `dblclick`
- Creates a utility that can be consumed in a developers projects where they manually generate wrappers (e.g. Ionic Framework generates manual wrappers for routing elements)

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Dev-build: `0.5.1-dev.11685564559.1b0b6e89`

Ionic Framework dev-build: `7.0.11-dev.11685565959.10c33d71`

### CODE REVIEW

**Review by rwaskiewicz — COMMENTED** [review]
One non-urgent question

File: packages/react-output-target/react-component-lib/utils/attachProps.ts [hunk 1]
```diff
@@ -62,6 +62,17 @@ export const getClassName = (classList: DOMTokenList, newProps: any, oldProps: a
   return finalClassNames.join(' ');
 };
 
+/**
+ * Transforms a React event name to a browser event name.
+ */
+export const transformReactEventName = (eventNameSuffix: string) => {
```
[the remark this answers is not in the document]
Comment (sean-perkins) [reply packages/react-output-target/react-component-lib/utils/attachProps.ts:68]: Yes, the event name comes from the JSX bindings which does not support a leading or trailing space as part of the property name. The usage that calls `isCoveredByReact` always converts the incoming event name to all lowercase.

File: packages/react-output-target/react-component-lib/utils/attachProps.ts [hunk 2]
```diff
@@ -62,6 +62,17 @@ export const getClassName = (classList: DOMTokenList, newProps: any, oldProps: a
...
+  switch (eventNameSuffix) {
+    case 'doubleclick':
+      return 'dblclick';
```
Comment (liamdebeasi) [inline packages/react-output-target/react-component-lib/utils/attachProps.ts:71]: The `doubleclick` vs `dblclick` naming is a React thing correct? So React devs need to use `onDoubleClick`?

**Review by liamdebeasi — APPROVED** [review]
One question, but otherwise looks good.

File: packages/react-output-target/react-component-lib/utils/attachProps.ts (hunk 2 above)
Comment (sean-perkins) [reply packages/react-output-target/react-component-lib/utils/attachProps.ts:71]: Correct, React developers would use `onDoubleClick`. This is an odd scenario where the JSX casing doesn't match the "standard" practice of aligning with the browser event name (e.g. `onClick` → `click`, `onInput` → `input`).

**Review by rwaskiewicz — APPROVED** [review]

File: packages/react-output-target/react-component-lib/utils/attachProps.ts (hunk 2 above)
Comment (liamdebeasi) [reply packages/react-output-target/react-component-lib/utils/attachProps.ts:71]: Thanks!

**Pull request merged**